beyondessential/tamanu
# [NASS-708] Create endpoint for exporting Reference data

### Changes
- Added new endpoint for exporting reference data


https://user-images.githubusercontent.com/17033058/231010181-df6a1a35-cbf3-4e6b-8dec-509bdee29a91.mov

### How to test
- Open desktop and point it to the sync server
- Login as admin
- Open Reference data
- Select export tab
- Select all the data you want to export
- Click to export
- It should create/download a file with all the reference data you selected
- Use this file to import into a new Sync server
- It should import all the data you exported previously.

### Checklist

- [ ] Code is finished
- [ ] Tests are written
- [ ] UI Screenshots added to the Linear issue
- [ ] Testing notes added to the Linear issue

_Upon merging:_

- [ ] Update the changelog (find it [here](https://github.com/beyondessential/tamanu/pulls?q=is%3Apr+is%3Aopen+label%3Arelease), or if the previous PR is merging into master then create a new release candidate PR for the next version following [the Slab doc](https://beyond-essential.slab.com/posts/merging-a-tamanu-ticket-okxp8s4s#h0y52-creating-a-release-candidate-pr))
  - [ ] with a ticket reference (e.g. `TAN-123: a one line description`, keep the lists sorted)
  - [ ] any config/settings changes and manual deployment steps
  - [ ] any db schema or other changes that could impact downstream data analysis
- [ ] Update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly) or [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q) as needed
- [ ] Update the [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c) as needed
